File: Dot examples/ASP Book Examples/Book Examples/Chapter34/vb/App_Code/CodeSamples/Chapter03/LengthValidator.vb
Imports System
Imports System.Web.UI
Imports System.Web.UI.WebControls

Namespace myControls

    ''' <summary>
    ''' Validates the length of an input field
    ''' </summary>
    Public Class LengthValidator
        Inherits BaseValidator

        Dim _maximumLength As Integer = 0

        Public Property MaximumLength() As Integer
            Get
                Return _maximumLength
            End Get
            Set(ByVal Value As Integer)
                _maximumLength = value
            End Set
        End Property

        Protected Overrides Function EvaluateIsValid() As Boolean
            Dim value As String = Me.GetControlValidationValue(Me.ControlToValidate)
            If value.Length > _maximumLength Then
                Return False
            Else
                Return True
            End If
        End Function
    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter34/cs/App_Code/CodeSamples/Chapter03/LengthValidator.cs
using System;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace CodeSamples.Chapter03
{
    /// <summary>
    /// Validates the length of an input field
    /// </summary>
    public class LengthValidator : BaseValidator
    {
        int _maximumLength = 0;

        public int MaximumLength
        {
            get { return _maximumLength; }
            set { _maximumLength = value; }
        }

        protected override bool EvaluateIsValid()
        {
            String value = this.GetControlValidationValue(this.ControlToValidate);
            if (value.Length > _maximumLength)
                return false;
            else
                return true;
        }
    }
}